cmd/aries-agent-lambda/aries.go

```diff
@@ -0,0 +1,387 @@
+	wsinternal "github.com/hyperledger/aries-framework-go/cmd/aries-agent-lambda/transport/apigw_ws"
+	arieshttp "github.com/hyperledger/aries-framework-go/pkg/didcomm/transport/http"
+	httpinternal "github.com/hyperledger/aries-framework-go/cmd/aries-agent-lambda/transport/apigw_http"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries"
+	ariescontext "github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/vdr/httpbinding"
+	"github.com/hyperledger/aries-framework-go/cmd/aries-agent-lambda/storage/dynamodb"
+)
+
+type dbParam struct {
+	dbType string
+	// prefix  string
+	// timeout uint64
+}
+
+type agentParameters struct {
+	// server                                         server
+	host, defaultLabel, transportReturnRoute       string
+	tlsCertFile, tlsKeyFile                        string
+	token, keyType, keyAgreementType               string
+	webhookURLs, httpResolvers, outboundTransports []string
+	inboundHostInternals, inboundHostExternals     []string
+	contextProviderURLs, mediaTypeProfiles         []string
+	autoAccept                                     bool
+	msgHandler                                     command.MessageHandler
+	dbParam                                        *dbParam
+	autoExecuteRFC0593                             bool
+}
+
+const (
+	httpProtocol      = "http"
+	websocketProtocol = "ws"
+	defaultWebhookUrl = "https://sly.slygo.scaffoldly.dev/aries-webhooks/api/v1/webhooks"
+)
+
+var (
+	keyTypes = map[string]kms.KeyType{
+		"ed25519":           kms.ED25519Type,
+		"ecdsap256ieee1363": kms.ECDSAP256TypeIEEEP1363,
+		"ecdsap256der":      kms.ECDSAP256TypeDER,
+		"ecdsap384ieee1363": kms.ECDSAP384TypeIEEEP1363,
+		"ecdsap384der":      kms.ECDSAP384TypeDER,
+		"ecdsap521ieee1363": kms.ECDSAP521TypeIEEEP1363,
+		"ecdsap521der":      kms.ECDSAP521TypeDER,
+	}
+
+	keyAgreementTypes = map[string]kms.KeyType{
+		"x25519kw": kms.X25519ECDHKWType,
+		"p256kw":   kms.NISTP256ECDHKWType,
+		"p384kw":   kms.NISTP384ECDHKWType,
+		"p521kw":   kms.NISTP521ECDHKWType,
+	}
+)
+
+func RegisterRoutes(router *mux.Router) (*mux.Router, error) {
+	// Tidbits from https://github.com/hyperledger/aries-framework-go/blob/9446ee81c3b4f39ca9f7af4abcf7cb4cc594e968/cmd/aries-agent-rest/startcmd/start.go#L736
+	dbParam := &dbParam{
+		dbType: "dynamodb",
+	}
+
+	httpProto := "https"
+	wsProto := "wss"
+	stage := os.Getenv("STAGE")
+	httpHost := os.Getenv("API_GATEWAY_DOMAIN")
+	wsHost := os.Getenv("API_GATEWAY_WEBSOCKET_DOMAIN")
+	internalHttpHost := httpHost
+	// internalWsHost := wsHost
+	basePath := os.Getenv("SERVICE_SLUG")
+	webhookPath := fmt.Sprintf("/%s/%s", basePath, "webhook")
+	if stage == "local" {
+		httpProto = "http"
+		wsProto = "ws"
+		httpHost = "localhost:3000"
+		wsHost = "localhost:3001"
+		internalHttpHost = "host.docker.internal:3000"
```

Comment:
We needed this hack at the time as Linux uses a proper bridge vs other platforms proxying on localhost. Possibly `serverless-offline` has mitigated the issue in the meantime.